tests/publiccloud/create_aistack_env.pm

```diff
@@ -0,0 +1,247 @@
+    assert_script_run("echo \"$ipaddr $host_name\" | sudo tee -a /etc/hosts > /dev/null");
+    set_var('OPENWEBUI_IP', "$ipaddr");
+    record_info("Added $ipaddr to /etc/hosts with hostname $host_name");
+    my $curl_cmd = "curl -v http://$host_name";
+    my $curl_result = script_run($curl_cmd);
+    if ($curl_result == 0) {
+        record_info("Successfully connected to the open-webui service at $curl_cmd \n");
+    } else {
+        die "Unable to connect to the open-webui service at $curl_cmd\n";
+    }
+}
+
+
+sub install_nvidia_drivers {
+    my ($instance, $values_url, $file_name) = @_;
+    my $easyinstall_url = 'https://gitlab.nue.suse.com/cloud-solutions-sys-eng/nvidia-drivers-easy-install/-/blob/main/nvidia_easy_install.sh';
+    record_info('Install nvidia drivers');
+    my $driver_version = '550.54.14';
+    #my $gpu_op_url = get_var('GPU_OPERATOR');
+    my $gpu_op_url = 'nvidia https://nvidia.github.io/gpu-operator';
+
+#    my $driver_version=script_output("zypper se -s nvidia-open-driver | grep nvidia-open-driver- | sed 's/.* package | //g' | sed 's/\s.*//g' | sort | head -n 1 | sed 's/[-_].*//g'");
+    script_run("sudo zypper ar https://download.nvidia.com/suse/sle15sp6/ nvidia-sle15sp6-main");
+    script_run("sudo zypper --gpg-auto-import-keys refresh");
+    script_run("sudo zypper install -y --auto-agree-with-licenses nvidia-open-driver-G06-signed-kmp=$driver_version nvidia-compute-utils-G06=$driver_version");
+
+    record_info('Install nvidia gpu operator');
+    assert_script_run("curl -o $file_name $values_url", timeout => 120);
+    assert_script_run("helm repo add $gpu_op_url", timeout => 600);
+    assert_script_run("helm repo update", timeout => 600);
+    assert_script_run("helm repo list", timeout => 600);
+    assert_script_run("helm install gpu-operator -n gpu-operator --create-namespace nvidia/gpu-operator --set driver.enabled=false -f $file_name ", timeout => 600);
+
+    #script_run("curl -sSL $easyinstall_url -o ./nvidia_easy_install.sh && chmod +x ./nvidia_easy_install.sh");
+    #script_run("sh ./nvidia_easy_install.sh");
+    #
+    #install nvidia gpu operator
+    #
+    process_reboot(trigger => 1);
+    script_run("sudo nvidia-smi");
+    script_run("kubectl get pods -n gpu-operator");
+}
+
+sub run {
+    my ($self, $args) = @_;
+    #my $values_url = 'https://raw.githubusercontent.com/yarunachalam/os-autoinst-distri-opensuse/aistack_basic/data/aistack/';
+    my $values_url = get_var('HELM_VALUES');
+    my $ai_ns = 'suse-private-ai';
+    my $value_file_name = '';
+
+    my $instance = $self->{my_instance} = $args->{my_instance};
+    my $provider = $self->{provider} = $args->{my_provider};
+
+    # Install dependency package
+    install_dependency_package($instance);
+    install_dependency_components($instance);
+    config_kubectl($instance);
+
+    if (get_var('PUBLIC_CLOUD_NVIDIA_GPU_AISTACK')) {
```

Comment:
You can check for a specific value, get_var will return undef, using checkvar makes way more explicit I guess
```suggestion
    if (check_var('PUBLIC_CLOUD_NVIDIA_GPU_AISTACK', 1)) {
```

Reply:
fixed